var/spack/repos/builtin/packages/py-meson-python/package.py

```diff
@@ -14,12 +14,22 @@ class PyMesonPython(PythonPackage):
 
     maintainers = ["eli-schwartz", "adamjstewart"]
 
+    version("0.10.0", sha256="08dd122c1074dbd5c55b53993a719cca73dd8216372c91217f7a550260f9e7e1")
+    version("0.9.0", sha256="6aa5a09ff5cce1c5308938ebbf3eab5529413c8677055ace1ac8c83d8a07b29d")
+    version("0.8.1", sha256="442f1fa4cf5db50eea61170a6059c10fafd70977f5dbdf3441c106cd23b05e4c")
+    version("0.8.0", sha256="b5c8a2727e6f6feaffc1db513244c9bdb5d0f689b45e24f4529b649b7710daf7")
     version("0.7.0", sha256="9fcfa350f44ca80dd4f5f9c3d251725434acf9a07d9618f382e6cc4629dcbe84")
 
     depends_on("python@3.7:", type=("build", "run"))
-    depends_on("py-meson@0.62:", type=("build", "run"))
+    depends_on("meson@0.63:", when="@0.9:", type=("build", "run"))
+    depends_on("meson@0.62:", type=("build", "run"))
     depends_on("py-ninja", type=("build", "run"))
```

Comment:
@adamjstewart can you remove this `py-ninja` dependency? It is not necessary, because `meson` itself already depends on `ninja`. And I noticed this because `py-ninja` pulls in a bunch of stuff, which also happens to break the build:
```
    ^py-meson-python@0.10.0%oneapi@2022.2.0 arch=linux-endeavourosrolling-skylake_avx512
        ^meson@0.63.3%oneapi@2022.2.0 patches=9a61825,aa6c50d arch=linux-endeavourosrolling-skylake_avx512
            ^ninja@1.10.2%oneapi@2022.2.0 arch=linux-endeavourosrolling-skylake_avx512
        ^py-ninja@1.10.2%oneapi@2022.2.0 arch=linux-endeavourosrolling-skylake_avx512
            ^cmake@3.24.2%oneapi@2022.2.0~doc+ncurses+ownlibs~qt build_type=Release arch=linux-endeavourosrolling-skylake_avx512
            ^py-scikit-build@0.15.0%oneapi@2022.2.0 arch=linux-endeavourosrolling-skylake_avx512
                ^py-distro@1.7.0%oneapi@2022.2.0 arch=linux-endeavourosrolling-skylake_avx512
                ^py-setuptools-scm@7.0.5%oneapi@2022.2.0+toml arch=linux-endeavourosrolling-skylake_avx512
                    ^py-typing-extensions@4.3.0%oneapi@2022.2.0 arch=linux-endeavourosrolling-skylake_avx512
```

Failure:
```
==> py-setuptools-scm: Executing phase: 'install'
==> Error: ProcessError: Command exited with status 1:
    '/home/rgommers/code/spack/opt/spack/linux-endeavourosrolling-skylake_avx512/oneapi-2022.2.0/python-3.9.13-mn4ovd4dkbyg62orinqikrnvaxq6r5tf/bin/python3.9' '-m' 'pip' '-vvv' '--no-input' '--no-cache-dir' '--disable-pip-version-check' 'install' '--no-deps' '--ignore-installed' '--no-build-isolation' '--no-warn-script-location' '--no-index' '--prefix=/home/rgommers/code/spack/opt/spack/linux-endeavourosrolling-skylake_avx512/oneapi-2022.2.0/py-setuptools-scm-7.0.5-hnb5qhebpnqeortsd2wrbwr4n76wbu4a' '.'

1 error found in build log:
     21      ╰─> See above for output.
     22    
     23      note: This error originates from a subprocess, and is likely not a problem with pip.
     24      full command: /home/rgommers/code/spack/opt/spack/linux-endeavourosrolling-skylake_avx512/oneapi-2022.2.0/python-3.9.13-mn4ovd4dkbyg62orinqikr
           nvaxq6r5tf/bin/python3.9 /home/rgommers/code/spack/opt/spack/linux-endeavourosrolling-skylake_avx512/oneapi-2022.2.0/py-pip-22.2.2-42kbxmhhxyigy
           p2224oarapc3o5xxpga/lib/python3.9/site-packages/pip/_vendor/pep517/in_process/_in_process.py prepare_metadata_for_build_wheel /tmp/tmpghv47zt8
     25      cwd: /tmp/rgommers/spack-stage/spack-stage-py-setuptools-scm-7.0.5-hnb5qhebpnqeortsd2wrbwr4n76wbu4a/spack-src
     26      Preparing metadata (pyproject.toml): finished with status 'error'
  >> 27    error: metadata-generation-failed
     28    
     29    × Encountered error while generating package metadata.
     30    ╰─> See above for output.
     31    
     32    note: This is an issue with the package mentioned above, not pip.
     33    hint: See above for details.

See build log for details:
  /tmp/rgommers/spack-stage/spack-stage-py-setuptools-scm-7.0.5-hnb5qhebpnqeortsd2wrbwr4n76wbu4a/spack-build-out.txt

==> Warning: Skipping build of py-scikit-build-0.15.0-4xj4s6zmrxin6ai6cruhlrlynyfrppz4 since py-setuptools-scm-7.0.5-hnb5qhebpnqeortsd2wrbwr4n76wbu4a failed
==> Warning: Skipping build of py-ninja-1.10.2-ccj33ijfb33sidxag3rrchkj6rup3edh since py-scikit-build-0.15.0-4xj4s6zmrxin6ai6cruhlrlynyfrppz4 failed
==> Warning: Skipping build of py-meson-python-0.10.0-2bju4amu52sa5spazhgj2wpbhsmz4xw6 since py-ninja-1.10.2-ccj33ijfb33sidxag3rrchkj6rup3edh failed
==> Warning: Skipping build of py-scipy-1.9.2-q4czoj4obdzhiagu4pmt7pbf7lqdarsc since py-meson-python-0.10.0-2bju4amu52sa5spazhgj2wpbhsmz4xw6 failed
[+] /home/rgommers/code/spack/opt/spack/linux-endeavourosrolling-skylake_avx512/oneapi-2022.2.0/py-build-0.7.0-qdhhhy6vg6iqphikberhszb3yh6tx2ve
==> Error: py-scipy-1.9.2-q4czoj4obdzhiagu4pmt7pbf7lqdarsc: Package was not installed
==> Error: Installation request failed.  Refer to reported errors for failing package(s).
```

The `py-ninja` dependency is never needed, Spack should always prefer regular `ninja`.